huaweicloud/services/evs/data_source_huaweicloud_evs_volumes_by_tags.go

```diff
@@ -0,0 +1,457 @@
+			"volume_image_metadata":        utils.PathSearch("volume_image_metadata", resourceDetail, map[string]interface{}{}),
+			"volume_type":                  utils.PathSearch("volume_type", resourceDetail, nil),
+			"size":                         utils.PathSearch("size", resourceDetail, nil),
+			"bootable":                     utils.PathSearch("bootable", resourceDetail, nil),
+			"metadata":                     utils.PathSearch("metadata", resourceDetail, map[string]interface{}{}),
+			"updated_at":                   utils.PathSearch("updated_at", resourceDetail, nil),
+			"service_type":                 utils.PathSearch("service_type", resourceDetail, nil),
+			"multiattach":                  utils.PathSearch("multiattach", resourceDetail, nil),
+			"dedicated_storage_id":         utils.PathSearch("dedicated_storage_id", resourceDetail, nil),
+			"dedicated_storage_name":       utils.PathSearch("dedicated_storage_name", resourceDetail, nil),
+			"tags":                         utils.PathSearch("tags", resourceDetail, map[string]interface{}{}),
+			"wwn":                          utils.PathSearch("wwn", resourceDetail, nil),
+			"enterprise_project_id":        utils.PathSearch("enterprise_project_id", resourceDetail, nil),
+		},
+	}
+}
+
+func flattenEvsLinks(links []interface{}) []map[string]interface{} {
+	rst := make([]map[string]interface{}, 0, len(links))
+	for _, link := range links {
+		rst = append(rst, map[string]interface{}{
+			"href": utils.PathSearch("href", link, nil),
+			"rel":  utils.PathSearch("rel", link, nil),
+		})
+	}
+	return rst
+}
+
+func flattenEvsAttachments(attachments []interface{}) []map[string]interface{} {
+	rst := make([]map[string]interface{}, 0, len(attachments))
+	for _, att := range attachments {
+		rst = append(rst, map[string]interface{}{
+			"server_id":     utils.PathSearch("server_id", att, nil),
+			"attachment_id": utils.PathSearch("attachment_id", att, nil),
+			"attached_at":   utils.PathSearch("attached_at", att, nil),
+			"volume_id":     utils.PathSearch("volume_id", att, nil),
+			"device":        utils.PathSearch("device", att, nil),
+			"id":            utils.PathSearch("id", att, nil),
+			"host_name":     utils.PathSearch("host_name", att, nil),
+		})
+	}
+	return rst
+}
+
+func dataSourceEvsVolumesByTagsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/cloudvolumes/resource_instances/action"
+		product = "evs"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	var allVolumes []interface{}
+	offset := 0
+	totalCount := 0
+
+	for {
+		requestBody := buildListEvsVolumesByTagsBody(d)
+		if requestBody["action"] == "filter" {
```

Comment:
It seems that action must and can only be a filter. This code needs to be optimized.